drivers/gpio/gpio_dw.c

```diff
@@ -205,28 +303,82 @@ static inline void dw_port_config(struct device *port, int flags)
 static inline int gpio_dw_config(struct device *port, int access_op,
 				 u32_t pin, int flags)
 {
+	const struct gpio_dw_config *config = port->config->config_info;
+	struct gpio_dw_runtime *context = port->driver_data;
+	u32_t port_base_addr = context->base_addr;
+	u32_t data_port = dw_get_data_port(port_base_addr);
+
+	/* Check for invalid pin number */
+	if (pin >= config->bits) {
+		return -EINVAL;
+	}
+
+	/* Does not support disconnected pin */
+	if (flags == GPIO_DISCONNECTED) {
```

Comment:
It would be better do to do something like:
```
u32 io_flags = flags & (GPIO_INPUT | GPIO_OUTPUT);
if ((io_flags == GPIO_DISCONNECTED)
    || (io_flags == (GPIO_INPUT | GPIO_OUTPUT))) {
    return -ENOTSUP;
}
```

There's no guarantee that somebody didn't provide other flags by mistake, and it's clearer if both unsupported configurations are handled at the same point in the code.